src/frontends/pytorch/src/frontend.cpp

```diff
@@ -40,41 +40,88 @@ namespace frontend {
 namespace pytorch {
 
 namespace {
-std::set<std::string> get_unconverted_types_from_model(const std::shared_ptr<Model>& model) {
-    std::set<std::string> unconverted_ops_types;
+std::map<std::string, std::string> get_unconverted_types_from_model(const std::shared_ptr<Model>& model) {
+    std::map<std::string, std::string> unconverted_ops_types;
     for (const auto& node : model->get_ordered_ops()) {
         if (const auto& fw_node = ov::as_type_ptr<PtFrameworkNode>(node)) {
-            auto attrs = fw_node->get_attrs();
-            FRONT_END_GENERAL_CHECK(attrs.find("PtTypeName") != attrs.end(),
+            const auto& attrs = fw_node->get_attrs();
+            FRONT_END_GENERAL_CHECK(attrs.find(PtFrameworkNode::op_type_key) != attrs.end(),
                                     "FrameworkNode attributes do not contain operation type.");
-            unconverted_ops_types.insert(attrs.at("PtTypeName"));
+            std::string exception_msg;
+            if (attrs.find(PtFrameworkNode::failed_conversion_key) != attrs.end()) {
+                exception_msg = attrs.at(PtFrameworkNode::failed_conversion_key);
+            }
+            if (!unconverted_ops_types.count(attrs.at(PtFrameworkNode::op_type_key))) {
+                unconverted_ops_types[attrs.at(PtFrameworkNode::op_type_key)] = exception_msg;
+            }
         }
         if (const auto& fw_node = ov::as_type_ptr<ov::op::util::MultiSubGraphOp>(node)) {
             for (size_t i = 0; i < fw_node->get_internal_subgraphs_size(); i++) {
-                auto internal_types = get_unconverted_types_from_model(fw_node->get_function(i));
+                const auto& internal_types = get_unconverted_types_from_model(fw_node->get_function(i));
                 unconverted_ops_types.insert(internal_types.begin(), internal_types.end());
             }
         }
     }
     return unconverted_ops_types;
 }
+
+std::string pack_detailed_failure_report(const std::map<std::string, std::string>& unconverted_ops,
+                                         const std::string& additional_error = "") {
+    std::stringstream error_msg;
+    std::stringstream unconverted_ops_msg;
+    std::stringstream failed_ops_msg;
+    std::stringstream failed_ops_short;
+    error_msg << "Model wasn't fully converted.";
+    unconverted_ops_msg << "-- No conversion rule found for operations: ";
+    failed_ops_msg << " Failed operations detailed log:";
+    failed_ops_short << "-- Conversion is failed for: ";
+    bool at_least_one = false;
+    bool at_least_one_except = false;
+    for (auto&& op : unconverted_ops) {
+        if (op.second.empty()) {
+            if (at_least_one)
+                unconverted_ops_msg << ", ";
```

Comment:
Will we have sorted list of unsupported ops?

Reply:
Yes